src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -136,26 +178,38 @@ const addCustomElementInputs = (buildCtx: d.BuildCtx, bundleOpts: BundleOptions)
 
     if (cmp.isPlain) {
       exp.push(`export { ${importName} as ${exportName} } from '${cmp.sourceFilePath}';`);
+      indexImps.push(`export { ${importName} as ${exportName} } from '${coreKey}';`);
     } else {
       // the `importName` may collide with the `exportName`, alias it just in case it does with `importAs`
       exp.push(
         `import { ${importName} as ${importAs}, defineCustomElement as cmpDefCustomEle } from '${cmp.sourceFilePath}';`
       );
       exp.push(`export const ${exportName} = ${importAs};`);
       exp.push(`export const defineCustomElement = cmpDefCustomEle;`);
+
+      // Here we push an import for this component onto our array which references the `coreKey` (prefixed with `\0`).
+      // We have to do this so that our import is referencing the correct virtual module, if we instead referenced, for
+      // instance, `cmp.sourceFilePath`, we would end up with duplicated modules in our output.
+      indexImps.push(
+        `import { ${importName} as ${importAs}, defineCustomElement as cmpDefCustomEle${exportName} } from '${coreKey}';`
```

Comment:
I think the comment explains it, but this was what I came up with to add an import to the `index.js` file without creating a problem with duplicated modules (which happens if you import from `cmp.sourceFilePath` directly)